src/routes/transactions/mappers/transfers/transfer-imitation.mapper.ts

```diff
@@ -113,7 +117,10 @@ export class TransferImitationMapper {
           return false;
         }
 
-        return this.isSpoofedEvent(txInfo, prevTxInfo);
+        return (
+          this.isSpoofedEvent(txInfo, prevTxInfo) ||
+          this.isEchoImitation(txInfo, prevTxInfo)
+        );
```

Comment:
(This is more a general comment, no directly related to the PR code, and I think it's out of the scope) 

As this `TransferImitationMapper` gets more complex, it is likely getting more computationally expensive. I think it's fine for now, but given this mapper is applied to each group of transactions, and if I'm not wrong it's deterministic, it'd be great to somehow cache/memoize whether a given transaction is an imitation one. Wdyt?

Reply:
I agree - we should look into caching the mapped entities (potentially across the project). I've noted it down for discussion.